### PR TITLE
Handle System.Nullable`1[System.Guid]

### LINKED ISSUE #1618: ASF-ui doesn't understand System.Nullable`1[System.Guid]
### Checklist

- [X] I read and understood ASF-ui's **[Contributing Guidelines](https://github.com/JustArchiNET/ASF-ui/blob/main/.github/CONTRIBUTING.md)**
- [X] I also read **[FAQ](https://github.com/JustArchiNET/ASF-ui/wiki/FAQ)**
- [X] This is not a **[question](https://github.com/JustArchiNET/ASF-ui/issues/new?assignees=&labels=%E2%9D%93+Question&template=Question.yml)**
- [X] This is not **[ASF problem](https://github.com/JustArchiNET/ArchiSteamFarm/issues/new/choose)**
- [X] I cleared my cache and localStorage

### ASF version

Latest pre-release

### ASF-ui version

81151ec

### Bug description

ASF 5.4.1.0 introduced new System.Nullable\`1[System.Guid] config property: `LicenseID`.

ASF-ui doesn't know how to handle it. It's nullable uuid, in json it's saved as text, such as `"LicenseID": "f6a0529813f74d119982eb4fe43a9a24"` - nullable means ASF also accepts `"LicenseID": null`.

Log:

```
InstallTrigger jest przestarzałe i zostanie usunięte w przyszłości. constants.js:50:14
i is null main.ccc0dfa.bundle.js:2:45130
Error: Failed!
    a http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    u http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    promise callback*l http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    promise callback*p http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    o http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    u http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    l http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    <anonymous> http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    u http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    u http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    created http://192.168.0.10:1242/scripts/4131.2aca93b.chunk.js:1
    Ce http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    tn http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    _init http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    a http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    componentInstance http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    init http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    n http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    d http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    d http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    C http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    C http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    C http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    C http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    $o http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    _update http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    r http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    get http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    run http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    dn http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    Re http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    Me http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    promise callback*je http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    Re http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    update http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    update http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    notify http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    set http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
    init http://192.168.0.10:1242/scripts/main.ccc0dfa.bundle.js:2
main.ccc0dfa.bundle.js:2:158443
```

### Expected behavior

ASF-ui should offer user a way to input `LicenseID`, minimum working example is treating it as `string`, but it'd be best if ASF-ui at least checked it against `/^[0-9a-fA-F]{8}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{12}$/` regex.

### Actual behavior

ASF-ui doesn't work with it at all, configuration page never loads.

### Steps to reproduce

Launch ASF 5.4.1.0 or newer, then try to open configuration section.

### Possible reason/solution

As per above.

### Can you help us with this bug report?

Somehow, I can test and offer feedback, but can't code

### Global ASF.json config file

_No response_

### BotName.json config of all affected bot instances

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by MrBurrBurr**
Fixes #1618

### CODE REVIEW

**Comment by JustArchi** [comment]
@MrBurrBurr It's working great!

Can we validate it against a regex of `^[0-9a-fA-F]{8}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{12}$`?

**Review by JustArchi — APPROVED** [review]
I tested this and it works great, but regex enhancement could make it even better.

**Comment by MrBurrBurr** [comment]
@JustArchi Any suggestions regarding the info texts?

File: src/i18n/locale/default.json [hunk 1]
```diff
@@ -73,7 +73,9 @@
   "commands-send": "Click here to send your command",
   "commands-default": "For a list of available commands type: commands\nFor a short description of a command type: help <command>",
   "config": "Config",
+  "config-invalid-guid": "LicenseID is not a valid guid!",
```
Comment (JustArchi) [inline src/i18n/locale/default.json:76]: I'd recommend either extracting `LicenseID` into a variable like `{property}`, or changing key to something like `config-invalid-license`.

**Review by JustArchi — APPROVED** [review]
LGTM

**Comment by JustArchi** [comment]
Thanks a lot for a very swift fix @MrBurrBurr :trophy:

File: src/utils/fetchConfigSchema.js [hunk 2]
```diff
@@ -53,6 +54,8 @@ async function resolveType(type) {
     case 'System.Collections.Generic.Dictionary':
     case 'System.Collections.Immutable.ImmutableDictionary':
       return { type: 'dictionary', key: await resolveType(subtypes[0]), value: await resolveType(subtypes[1]) };
+    case 'System.Nullable':
+      return { type: 'nullable', values: await resolveType(subtypes[0]) };
```
Comment (Aareksio) [inline src/utils/fetchConfigSchema.js:58]: Shouldn't it be `{ type: await resolveType(subtypes[0]), nullable: true }` or similar?

File: src/components/Config/Editor.vue [hunk 3]
```diff
@@ -90,6 +90,8 @@
         switch (field.type) {
           case 'string':
           case 'uint64':
+          case 'guid':
+          case 'nullable':
```
Comment (Aareksio) [inline src/components/Config/Editor.vue:94]: Because there is no reason to map `nullable` to `string`

File: src/utils/fetchConfigSchema.js (hunk 2 above)
Comment (MrBurrBurr) [reply src/utils/fetchConfigSchema.js:58]: Jup, thank you. Fixed in https://github.com/JustArchiNET/ASF-ui/commit/ad40d8e1324e34cb4f6ed7cd1a227ead03c99f06

File: src/components/Config/Editor.vue (hunk 3 above)
Comment (MrBurrBurr) [reply src/components/Config/Editor.vue:94]: Fixed in https://github.com/JustArchiNET/ASF-ui/commit/ad40d8e1324e34cb4f6ed7cd1a227ead03c99f06

**Pull request merged**